core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -115,4 +126,103 @@ public String getLoadBalancerUserFacingFQDN(Long stackId) {
         }
         return loadBalancerOptional;
     }
+
+    public Set<LoadBalancer> createLoadBalancers(Stack stack, DetailedEnvironmentResponse environment) {
+        LOGGER.info("Setting up load balancers for stack {}", stack.getDisplayName());
+        Set<LoadBalancer> loadBalancers = new HashSet<>();
+
+        if (isSupportedStackType(stack.getType()) && isLoadBalancerEnabled(environment)) {
+            LOGGER.debug("Load balancers are enabled for data lake and data hub stacks.");
+            Optional<TargetGroup> knoxTargetGroup = setupKnoxTargetGroup(stack);
+            if (knoxTargetGroup.isPresent()) {
+                if (arePrivateSubnetsAvailable(environment.getNetwork())) {
+                    LOGGER.debug("Found Knox enabled instance groups in stack. Setting up internal Knox load balancer");
+                    setupKnoxLoadBalancer(
+                        createLoadBalancerIfNotExists(loadBalancers, LoadBalancerType.PRIVATE, stack),
+                        knoxTargetGroup.get());
+                }
+                if (shouldCreateExternalKnoxLoadBalancer(environment.getNetwork())) {
+                    LOGGER.debug("Public endpoint access gateway is enabled. Setting up public Knox load balancer");
+                    setupKnoxLoadBalancer(
+                        createLoadBalancerIfNotExists(loadBalancers, LoadBalancerType.PUBLIC, stack),
+                        knoxTargetGroup.get());
+                }
+            } else {
+                LOGGER.debug("No Knox instance groups found. If load balancer creation is enabled, Knox routing in the load balancer will be skipped.");
+            }
+
+            // TODO CB-9368 - create target group for CM instances
+        }
+
+        return loadBalancers;
+    }
+
+    private boolean isLoadBalancerEnabled(DetailedEnvironmentResponse environment) {
+        return environment != null &&
+            (entitlementService.datalakeLoadBalancerEnabled(ThreadBasedUserCrnProvider.getAccountId()) ||
```

Comment:
This part of the condition should check also the stack type, shouldn't it?
If datalakeLoadBalancerEnabled returns with true then a DH cluster evaluates to true and will have load balancer configured.

Reply:
Thanks for that catch! I've changed the entitlement checks to be dependent on the stack type, and have also added a unit test case for a datahub stack with only the datalakeLoadBalancerEnabled entitlement, to make sure we don't create load balancers in that case.